src/openrct2-ui/windows/LandRights.cpp

```diff
@@ -45,409 +46,361 @@ static rct_widget window_land_rights_widgets[] = {
+                    show_land_rights();
+                    Invalidate();
+                }
+                break;
+            case WIDX_BUY_CONSTRUCTION_RIGHTS:
+                if (_landRightsMode != LAND_RIGHTS_MODE_BUY_CONSTRUCTION_RIGHTS)
+                {
+                    tool_set(this, WIDX_BUY_CONSTRUCTION_RIGHTS, Tool::UpArrow);
+                    _landRightsMode = LAND_RIGHTS_MODE_BUY_CONSTRUCTION_RIGHTS;
+                    show_construction_rights();
+                    Invalidate();
+                }
+                break;
+        }
     }
-}
 
-static void WindowLandRightsMousedown(rct_window* w, rct_widgetindex widgetIndex, rct_widget* widget)
-{
-    switch (widgetIndex)
+    void OnMouseDown(rct_widgetindex widgetIndex) override
     {
-        case WIDX_DECREMENT:
-            // Decrement land rights tool size
-            gLandToolSize = std::max(MINIMUM_TOOL_SIZE, gLandToolSize - 1);
-
-            // Invalidate the window
-            w->Invalidate();
-            break;
-        case WIDX_INCREMENT:
-            // Decrement land rights tool size
-            gLandToolSize = std::min(MAXIMUM_TOOL_SIZE, gLandToolSize + 1);
-
-            // Invalidate the window
-            w->Invalidate();
-            break;
+        switch (widgetIndex)
+        {
+            case WIDX_DECREMENT:
+                // Decrement land rights tool size
+                gLandToolSize = std::max(MINIMUM_TOOL_SIZE, gLandToolSize - 1);
+
+                // Invalidate the window
+                Invalidate();
+                break;
+            case WIDX_INCREMENT:
+                // Decrement land rights tool size
+                gLandToolSize = std::min(MAXIMUM_TOOL_SIZE, gLandToolSize + 1);
+
+                // Invalidate the window
+                Invalidate();
+                break;
+        }
     }
-}
 
-static void WindowLandRightsTextinput(rct_window* w, rct_widgetindex widgetIndex, char* text)
-{
-    int32_t size;
-    char* end;
-
-    if (widgetIndex != WIDX_PREVIEW || text == nullptr)
-        return;
-
-    size = strtol(text, &end, 10);
-    if (*end == '\0')
+    void OnTextInput(rct_widgetindex widgetIndex, std::string_view text) override
     {
-        size = std::max(MINIMUM_TOOL_SIZE, size);
-        size = std::min(MAXIMUM_TOOL_SIZE, size);
-        gLandToolSize = size;
-        w->Invalidate();
-    }
-}
+        if (text.empty())
+            return;
 
-static void WindowLandRightsInputsize(rct_window* w)
-{
-    Formatter ft;
-    ft.Add<int16_t>(MINIMUM_TOOL_SIZE);
-    ft.Add<int16_t>(MAXIMUM_TOOL_SIZE);
-    WindowTextInputOpen(w, WIDX_PREVIEW, STR_SELECTION_SIZE, STR_ENTER_SELECTION_SIZE, ft, STR_NONE, STR_NONE, 3);
-}
+        if (widgetIndex != WIDX_PREVIEW)
+            return;
 
-static void WindowLandRightsUpdate(rct_window* w)
-{
-    w->frame_no++;
-    // Close window if another tool is open
-    if (!LandRightsToolIsActive())
-        window_close(w);
-}
+        int32_t size;
```

Comment:
no need to declare this here. Move it to its first use on L159

Reply:
Fixed.